front/components/items/AppointmentApproval/index.tsx

```diff
@@ -0,0 +1,146 @@
+                            }`}
+                            colorClass="greendark"
+                            placeClass="center"
+                        />
+                        {/* TODO print appointment_code */}
+                        {[appointmentID]?.map((item, index) => {
+                            return (
+                                <React.Fragment key={index}>
+                                    <Paragraph
+                                        size="medium"
+                                        text={
+                                            'Wizyta u ' +
+                                            item?.attributes.therapist.data.attributes.first_name +
+                                            ' ' +
+                                            item?.attributes.therapist.data.attributes.last_name +
+                                            ' w dniu ' +
+                                            item?.attributes.date.split('-').reverse().join('.') +
+                                            'r. o godzinie ' +
+                                            item?.attributes.time.slice(0, 5)
+                                        }
+                                        colorClass="greendark"
+                                    />
+                                    <Paragraph
+                                        size="medium"
+                                        text={`${
+                                            (isApproved &&
+                                                'Twój kod wizyty to ' +
+                                                    findProperAppointmentCode({
+                                                        appointments,
+                                                        appointmentID,
+                                                    })) ||
+                                            (isDeclined && ' ') ||
+                                            'Koszt: ' +
+                                                item?.attributes.therapist.data.attributes
+                                                    .session_cost +
+                                                'zł. Płatność przed wizytą gotówką lub kartą'
+                                        }`}
+                                        colorClass="greendark"
+                                    />
+                                </React.Fragment>
+                            );
+                        })}
+
+                        <div className="appointmentapproval-alert__buttons">
+                            {!isApproved && !isDeclined ? (
+                                <React.Fragment>
+                                    <Button
+                                        variant="h3"
+                                        colorClass="greendark"
+                                        text="Zatwierdź"
+                                        onClick={() => {
```

Comment:
think about 
`await () => { await strapiAppointmentPut ....`

Idk if its needed, but it sounds safer for me.
But dont rewrite it without considering what is the better option